releases/release-5.0.4.md

```diff
@@ -0,0 +1,214 @@
+
++ PD
+
+    - Fix the bug that PD would not fix down-peer in time. [#4082](https://github.com/tikv/pd/pull/4082)
+    - Fix an issue where data is not stored when using max-replicas or location-labels to indirectly update default placement rule [#3914](https://github.com/tikv/pd/pull/3914)
+    - Fix the bug that PD may panic during scaling out TiKV. [#3910](https://github.com/tikv/pd/pull/3910)
+    - Reduce the conflict due to multiple scheduler running in same time [#3856](https://github.com/tikv/pd/pull/3856)
+    - Fix the issue that the scheduler may appear again even if we have already executed the delete operation [#3823](https://github.com/tikv/pd/pull/3823)
+
++ TiFlash
+
+    - Fix the potential panic issue that occurs when running table scan tasks
+    - Fix the potential memory leak issue that occurs when executing `MPP` tasks
+    - Fix a bug that TiFlash raises the error about `duplicated region` when handling `DAQ` requests
+    - Fix the issue of unexpected results when executing the aggregation functions `COUNT` or `COUNT DISTINCT`
+    - Fix the potential panic issue that occurs when executing `MPP` tasks
+    - Fix a potential bug that TiFlash cannot restore data when deployed on multiple disks
+    - Fix the potential panic issue that occurs when deconstructing `SharedQueryBlockInputStream`
+    - Fix the potential panic issue that occurs when deconstructing `MPPTask`
+    - Fix the issue of unexpected results when TiFlash failed to establish `MPP` connections
+    - Fix the potential panic issue that occurs when resolving locks
+    - Fix the issue that store size in metrics is inaccurate under heavy writing
+    - Fix a bug of incorrect results that occurs when queries contain filters like `CONSTANT` `<` | `<=` | `>` | `>=` `COLUMN`
+    - Fix the potential issue that TiFlash cannot GC the delta data after running for a long time
+    - Fix a potential bug that metrics display wrong value
+    - Fix the potential issue of data inconsistency that occurs when deployed on multiple disks
+
++ PD
+
+    - Fix the bug that PD would not fix down-peer in time. [#4082](https://github.com/tikv/pd/pull/4082)
+    - Fix an issue where data is not stored when using max-replicas or location-labels to indirectly update default placement rule [#3914](https://github.com/tikv/pd/pull/3914)
+    - Fix the bug that PD may panic during scaling out TiKV. [#3910](https://github.com/tikv/pd/pull/3910)
+
++ Tools
+
+    + Dumpling
+
+        - fix pending on show table status in some mysql version [#342](https://github.com/pingcap/dumpling/pull/342)
+        - Support for backing up MySQL compatible databases that don't support START TRANSACTION  ... WITH CONSISTENT SNAPSHOT
+        - Support for backing up MySQL compatible databases that don't support SHOW CREATE TABLE [#327](https://github.com/pingcap/dumpling/pull/327)
+
+    + TiCDC
+
```

Comment:
- Fix json encoding could panic when processing a string type value in some cases. [#2782](https://github.com/pingcap/ticdc/pull/2782)
        - Fix a bug that multiple processors could write the same table when this table is re-scheduling [#2728](https://github.com/pingcap/ticdc/pull/2728)
        - Fix OOM when TiCDC captures too many regions [#2724](https://github.com/pingcap/ticdc/pull/2724)
        - Fix gRPC keepalive error when memory pressure is high. [#2719](https://github.com/pingcap/ticdc/pull/2719)
        - Fix a bug that causes TiCDC to panic on an unsigned tinyint [#2655](https://github.com/pingcap/ticdc/pull/2655)
        - Fix open protocol, don't output an empty value when there is no change in one transaction. [#2620](https://github.com/pingcap/ticdc/pull/2620)
        - Fixed a bug in DDL handling when the owner restarts. [#2610](https://github.com/pingcap/ticdc/pull/2610)
        - Fix a bug in metadata management [#2558](https://github.com/pingcap/ticdc/pull/2558)
        - Add a global gRPC connection pool and share gRPC connections among kv clients. [#2533](https://github.com/pingcap/ticdc/pull/2533)
        - Fix a bug that multiple processors could write the same table when this table is re-scheduling [#2492](https://github.com/pingcap/ticdc/pull/2492)
        - fix outdated capture info may appear in capture list command [#2466](https://github.com/pingcap/ticdc/pull/2466)
        - Fix a bug that owner could meet ErrSchemaStorageTableMiss error and reset a changefeed by accident. [#2458](https://github.com/pingcap/ticdc/pull/2458)
        - fix the bug that changefeed cannot be removed if meet GcTTL Exceeded Error [#2456](https://github.com/pingcap/ticdc/pull/2456)
        - fix a bug where synchronizing large tables to cdclog failed. [#2445](https://github.com/pingcap/ticdc/pull/2445)
        - Fix CLI back-compatibility [#2413](https://github.com/pingcap/ticdc/pull/2413)
        - Fix minor runtime panic risk [#2299](https://github.com/pingcap/ticdc/pull/2299)
        - Fix potential DDL loss when owner crashes while executing DDL [#2292](https://github.com/pingcap/ticdc/pull/2292)
        - Fix extra partition dispatching when adding new table partition. [#2263](https://github.com/pingcap/ticdc/pull/2263)